src/MemoryOptions.php

```diff
@@ -75,18 +75,22 @@ public function getMemoryLimit()
     /**
      * Normalized a given value of memory limit into the number of bytes
      *
-     * @param string|int $value
+     * @param string|int|null $value
+     * @return int|null
      * @throws Exception\InvalidArgumentException
-     * @return int
      */
     protected function normalizeMemoryLimit($value)
     {
+        if ($value === null) {
+            return null;
+        }
```

Comment:
I believe this never accepted null. It would always fail on regex failing to match

Reply:
When I remove the line, the existing test `testOptionsFluentInterface` is failing, as preg_match does not accept `null` but `string` as as argument. It appears this test is trying to set `null` as the current value is `null`. The test looks if setting the current value leads to the same result which is already set.

`1) LaminasTest\Cache\Storage\Adapter\MemoryOptionsTest::testOptionsFluentInterface
TypeError: preg_match(): Argument #2 ($subject) must be of type string, null given

/var/www/html/library/laminas-cache-storage-adapter-memory/src/MemoryOptions.php:92
/var/www/html/library/laminas-cache-storage-adapter-memory/src/MemoryOptions.php:41
/var/www/html/library/laminas-cache-storage-adapter-memory/vendor/laminas/laminas-cache-storage-adapter-test/src/AbstractCommonAdapterTest.php:160`

Reply:
I think we should cast the value to `(string)`, then let `Exception\InvalidArgumentException` be thrown (below)

Reply:
But how to fix the test from the common base test class?

Reply:
It's an invalid option, no? If there is a way to set it to invalid, we'd reach this scenario.

The best course of action would be to completely disallow it being an invalid value in first place :)

Reply:
type error from null passed to preg_match is new in php 8 where native types were added to built-in functions.

The type error is due to missing argument type validation. 
Proper approach would be to check if value is not an int an not a string then throw invalid argument exception.

Reply:
If I add this instead of null check: 

```
if(! is_string($value) && ! is_int($value)) {
    throw new Exception\InvalidArgumentException('Memory limit must be an integer or string');
}
```

psalm is crying about:

```
ERROR: DocblockTypeContradiction - src/MemoryOptions.php:84:12 - Docblock-defined type int for $value is always int (see https://psalm.dev/155)
        if(! is_string($value) && ! is_int($value)) {
```

Furthermore it still breaks the test `testOptionsFluentInterface` as the test is not expecting the new Exception `Memory limit must be an integer or string`.